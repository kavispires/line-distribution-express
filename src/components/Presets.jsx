import React, { useCallback } from 'react';

import useGlobalState from '../useGlobalState';
import { SCREENS } from '../utils/constants';
import { buildActiveGroup } from '../utils';

function Presets() {
  // Global States
  const [, setScreen] = useGlobalState('screen');
  const [activeGroup, setActiveGroup] = useGlobalState('activeGroup');
  const [presets] = useGlobalState('presets');
  const [customPresets] = useGlobalState('customPresets');
  const [, setLog] = useGlobalState('log');

  const activatePreset = useCallback(
    (event) => {
      const { id } = event.target;
      if (id && id !== activeGroup?.id) {
        setActiveGroup(buildActiveGroup(event.target.id));
        // Reset
        setLog([]);
      }
      setScreen(SCREENS.DISTRIBUTE);
    },
    [activeGroup, setScreen, setActiveGroup, setLog]
  );

  return (
    <main className="content preset">
      <h2>Presets</h2>
      <p>Select the group preset you want to use:</p>
      <ul className="presets-list">
        {presets.map((preset) => {
          return (
            <li key={preset.id} className="preset-item">
              <button className="preset-button" id={preset.id} onClick={activatePreset}>
                {preset.name} ({preset.groupSize} members)
              </button>
            </li>
          );
        })}
      </ul>
      <h3>Custom Presets</h3>

      {customPresets.length > 0 ? (
        <ul className="presets-list">
          {presets.map((preset) => {
            return (
              <li key={preset.id} className="ha">
                {preset.name} ({preset.groupSize})
              </li>
            );
          })}
        </ul>
      ) : (
        <p>You have no custom presets saved on this browser.</p>
      )}
    </main>
  );
}

export default Presets;
