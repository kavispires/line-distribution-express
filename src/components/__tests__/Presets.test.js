import React from 'react';
import { render } from '@testing-library/react';
import Presets from '../Presets';

describe('components/Preset', () => {
  test('it renders', () => {
    const component = render(<Presets />);
    expect(component).toMatchSnapshot();
  });

  test('its title is correct', () => {
    const { getByTestId } = render(<Presets />);

    expect(getByTestId('presets-title')).toHaveTextContent('Presets');
  });

  test('its presets buttons are correct', () => {
    const { getByTestId } = render(<Presets />);
    console.log(getByTestId('presets-list'));
    expect(getByTestId('presets-list').child).toBe('Presets');
  });
});
