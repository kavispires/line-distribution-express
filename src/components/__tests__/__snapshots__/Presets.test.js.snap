// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Preset it renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="content preset"
      >
        <h2
          data-testid="presets-title"
        >
          Presets
        </h2>
        <p>
          Select the group preset you want to use:
        </p>
        <ul
          class="presets-list"
          data-testid="presets-list"
        >
          <li
            class="preset-item"
          >
            <button
              class="preset-button"
              id="a005"
            >
              Banda Vyrus
               (
              6
               members)
            </button>
          </li>
          <li
            class="preset-item"
          >
            <button
              class="preset-button"
              id="a001"
            >
              Squirrel Zone
               (
              11
               members)
            </button>
          </li>
          <li
            class="preset-item"
          >
            <button
              class="preset-button"
              id="a003"
            >
              Squirrel Zone 1.5
               (
              5
               members)
            </button>
          </li>
          <li
            class="preset-item"
          >
            <button
              class="preset-button"
              id="b002"
            >
              Squirrel Zone I
               (
              5
               members)
            </button>
          </li>
          <li
            class="preset-item"
          >
            <button
              class="preset-button"
              id="a004"
            >
              Squirrel Zone II
               (
              5
               members)
            </button>
          </li>
        </ul>
        <h3>
          Custom Presets
        </h3>
        <p>
          You have no custom presets saved on this browser.
        </p>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="content preset"
    >
      <h2
        data-testid="presets-title"
      >
        Presets
      </h2>
      <p>
        Select the group preset you want to use:
      </p>
      <ul
        class="presets-list"
        data-testid="presets-list"
      >
        <li
          class="preset-item"
        >
          <button
            class="preset-button"
            id="a005"
          >
            Banda Vyrus
             (
            6
             members)
          </button>
        </li>
        <li
          class="preset-item"
        >
          <button
            class="preset-button"
            id="a001"
          >
            Squirrel Zone
             (
            11
             members)
          </button>
        </li>
        <li
          class="preset-item"
        >
          <button
            class="preset-button"
            id="a003"
          >
            Squirrel Zone 1.5
             (
            5
             members)
          </button>
        </li>
        <li
          class="preset-item"
        >
          <button
            class="preset-button"
            id="b002"
          >
            Squirrel Zone I
             (
            5
             members)
          </button>
        </li>
        <li
          class="preset-item"
        >
          <button
            class="preset-button"
            id="a004"
          >
            Squirrel Zone II
             (
            5
             members)
          </button>
        </li>
      </ul>
      <h3>
        Custom Presets
      </h3>
      <p>
        You have no custom presets saved on this browser.
      </p>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components/Preset its title is correct 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="content preset"
      >
        <h2
          data-testid="presets-title"
        >
          Presets
        </h2>
        <p>
          Select the group preset you want to use:
        </p>
        <ul
          class="presets-list"
          data-testid="presets-list"
        >
          <li
            class="preset-item"
          >
            <button
              class="preset-button"
              id="a005"
            >
              Banda Vyrus
               (
              6
               members)
            </button>
          </li>
          <li
            class="preset-item"
          >
            <button
              class="preset-button"
              id="a001"
            >
              Squirrel Zone
               (
              11
               members)
            </button>
          </li>
          <li
            class="preset-item"
          >
            <button
              class="preset-button"
              id="a003"
            >
              Squirrel Zone 1.5
               (
              5
               members)
            </button>
          </li>
          <li
            class="preset-item"
          >
            <button
              class="preset-button"
              id="b002"
            >
              Squirrel Zone I
               (
              5
               members)
            </button>
          </li>
          <li
            class="preset-item"
          >
            <button
              class="preset-button"
              id="a004"
            >
              Squirrel Zone II
               (
              5
               members)
            </button>
          </li>
        </ul>
        <h3>
          Custom Presets
        </h3>
        <p>
          You have no custom presets saved on this browser.
        </p>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="content preset"
    >
      <h2
        data-testid="presets-title"
      >
        Presets
      </h2>
      <p>
        Select the group preset you want to use:
      </p>
      <ul
        class="presets-list"
        data-testid="presets-list"
      >
        <li
          class="preset-item"
        >
          <button
            class="preset-button"
            id="a005"
          >
            Banda Vyrus
             (
            6
             members)
          </button>
        </li>
        <li
          class="preset-item"
        >
          <button
            class="preset-button"
            id="a001"
          >
            Squirrel Zone
             (
            11
             members)
          </button>
        </li>
        <li
          class="preset-item"
        >
          <button
            class="preset-button"
            id="a003"
          >
            Squirrel Zone 1.5
             (
            5
             members)
          </button>
        </li>
        <li
          class="preset-item"
        >
          <button
            class="preset-button"
            id="b002"
          >
            Squirrel Zone I
             (
            5
             members)
          </button>
        </li>
        <li
          class="preset-item"
        >
          <button
            class="preset-button"
            id="a004"
          >
            Squirrel Zone II
             (
            5
             members)
          </button>
        </li>
      </ul>
      <h3>
        Custom Presets
      </h3>
      <p>
        You have no custom presets saved on this browser.
      </p>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
